### DISCUSSION TITLE
Parsing help output

### DISCUSSION THREAD

**Discussion by kitschpatrol** [kellyjonbrazil/jc#565] [Ideas]
It can be useful to parse the output of a command line tool's help (`--help` / `-h`) to generate documentation.

Of course this is a bit tricky since help output isn't consistently formatted, many tools have subcommands, etc., though one could probably cover a majority of tools by implementing parsers for the output of the most popular CLI frameworks (e.g. Yargs, Click, etc.)

I've partially implemented this via Chevrotain for [another project](https://github.com/kitschpatrol/mdat/tree/main/packages/mdat) (before discovering jc), with [support for parsing Yargs and Meow help output](https://github.com/kitschpatrol/mdat/tree/main/packages/mdat/src/lib/readme/rules/cli-help/utilities/parsers).   

This is ultimately used to automatically generate Markdown tables enumerating available commands and options of various tools in readme documentation.

Just a thought! Thanks for a brilliant tool.

**Comment by kellyjonbrazil** [comment]
Hi! That is an interesting idea... I was using the help text converter in zsh when first playing with autocomplete, so there may be some other cool use cases.
